### PR TITLE
Update Table

### PR DESCRIPTION
**Pull request by Ogaga01**
## Hello Reviewer 👋 
**In this project milestone, I implemented the following requirements:**
- Update  animals Table
- Delete animals Table
- Rollback animals Table
- use Transaction
- use savepoints
- Rollback to savepoints
- Use Aggregate
- use group by
Screenshot (56)
Screenshot (57)
Screenshot (58)
Screenshot (59)
Screenshot (60)
Screenshot (61)
Screenshot (62)
Screenshot (63)
Screenshot (64)
Screenshot (65)
Screenshot (66)
Screenshot (67)

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -3,4 +3,48 @@
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Agumon', '2020-02-03', 0, '1', 10.23);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Gabumon', '2018-11-15', 2, '1', 8);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu', '2021-01-07', 1, '0', 15.04);
- INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', '2017-05-12', 5, '1', 11);
\ No newline at end of file
+ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', '2017-05-12', 5, '1', 11);
+ ALTER TABLE animals ADD species VARCHAR(25);
```
Comment (DeliceLydia) [inline data.sql:7]: - [x] This should be located in `schema.sql` not here please address that

File: data.sql [hunk 2]
```diff
@@ -3,4 +3,48 @@
...
+ ('Blossom', '1998-10-13', 3, '1', 17), 
+ ('Ditto', '2022-05-14', 4, '1', 22);
+
+ BEGIN;
+ UPDATE animals SET species = 'unspecified';
+ COMMIT;
+
+ BEGIN;
+ UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+ UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
```
Comment (DeliceLydia) [inline data.sql:23]: - [x] I suggest that before `commit` you verify if the change was made you can do:
```
SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

File: data.sql [hunk 3]
```diff
@@ -3,4 +3,48 @@
...
+ COMMIT;
+
+ BEGIN;
+ DROP TABLE;
+ ROLLBACK;
+
+ BEGIN;
+ DELETE FROM animals WHERE date_of_birth > '2022-01-01';
```
Comment (DeliceLydia) [inline data.sql:31]: - Because you are deleting animals born after Jan 1st, 2022 you can do `date_of_birth >= '2022-01-01`

File: data.sql [hunk 4]
```diff
@@ -3,4 +3,48 @@
...
+
+SELECT AVG(weight_kg) FROM animals;
+
+SELECT COUNT(escape_attempts) FROM animals GROUP BY neutered;
+
+SELECT MAX(weight_kg) FROM animals GROUP BY species;
+
+SELECT MIN(weight_kg) FROM animals GROUP BY species;
+
+SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```
Comment (DeliceLydia) [inline data.sql:50]: - [x] Please remember that all queries should be located in `queries.sql` file please move them

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Ogaga01, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! highlights for your project:
- Followed git workflow
- Awesome README


## Required Changes ♻️
_Check the comments under the review._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: data.sql (hunk 2 above)
Comment (Ogaga01) [reply data.sql:23]: @DeliceLydia I realized this after I had already committed the changes, but it has been corrected now

File: data.sql (hunk 3 above)
Comment (Ogaga01) [reply data.sql:31]: @DeliceLydia Done

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @Ogaga01 

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**